snowplow-demo-app/src/main/java/com/snowplowanalytics/snowplowtrackerdemo/Demo.java

```diff
@@ -182,43 +185,132 @@ public void onCheckedChanged(final RadioGroup radioGroup, final int i) {
         _startButton.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 if (Build.VERSION.SDK_INT < 24) {
-                    setupAndTrackDemoEvents();
+                    setupTracker(new androidx.core.util.Consumer<Boolean>() {
+                        @Override
+                        public void accept(Boolean aBoolean) {
+                            trackEvents();
+                        }
+                    });
                 }
                 requestPermissions((isGranted) -> {
                     if (isGranted) {
-                        setupAndTrackDemoEvents();
+                        setupTracker(new androidx.core.util.Consumer<Boolean>() {
+                            @Override
+                            public void accept(Boolean aBoolean) {
+                                trackEvents();
+                            }
+                        });
+                        trackEvents();
                     }
                 });
             }
         });
     }
 
-    private void setupAndTrackDemoEvents() {
-        String uri = _uriField.getText().toString();
+    // Configuration
+
+    private void setupTracker(@NonNull androidx.core.util.Consumer<Boolean> callbackTrackerReady) {
+        boolean isRemoteConfig = _remoteConfig.getCheckedRadioButtonId() == _radioRemoteConfig.getId();
+        if (isRemoteConfig) {
+            setupWithRemoteConfig(callbackTrackerReady);
+        } else {
+            setupWithLocalConfig();
+        }
+    }
 
+    private boolean setupWithRemoteConfig(@NonNull androidx.core.util.Consumer<Boolean> callbackTrackerReady) {
+        String uri = _uriField.getText().toString();
+        if (uri.isEmpty()) {
+            uri = "https://mobile-app-config-bucket.s3.us-east-2.amazonaws.com/materialistic-v2-config.json";
```

Comment:
We probably don't want the default remote config in the example to be one we are hosting, someone will probably start using it by accident and I don't think we need to host it. I think I'd just make it error if the uri isn't valid or if it's empty. You could also put this file in a /config folder in the app and leave a comment here (or update the error message) to point people towards to example in the /config folder that needs hosting somewhere.